server/src/main/java/org/opensearch/action/search/CreatePITResponse.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.action.ActionResponse;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.StatusToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.rest.action.RestActions;
+
+import java.io.IOException;
+
+/**
+ * Create point in time response with point in time id and shard success / failures
+ */
+public class CreatePITResponse extends ActionResponse implements StatusToXContentObject {
+    // point in time id
+    private final String id;
+    private final int totalShards;
+    private final int successfulShards;
+    private final int failedShards;
+    private final int skippedShards;
+    private final ShardSearchFailure[] shardFailures;
+    private final long creationTime;
+
+    public CreatePITResponse(SearchResponse searchResponse, long creationTime) {
+        if (searchResponse.pointInTimeId() == null || searchResponse.pointInTimeId().isEmpty()) {
+            throw new IllegalArgumentException("Point in time ID is empty");
+        }
+        this.id = searchResponse.pointInTimeId();
+        this.totalShards = searchResponse.getTotalShards();
+        this.successfulShards = searchResponse.getSuccessfulShards();
+        this.failedShards = searchResponse.getFailedShards();
+        this.skippedShards = searchResponse.getSkippedShards();
+        this.shardFailures = searchResponse.getShardFailures();
+        this.creationTime = creationTime;
+    }
+
+    public CreatePITResponse(StreamInput in) throws IOException {
+        super(in);
+        id = in.readString();
```

Comment:
It seems like the `read` and `write` parts are completely out of sync [1]?

```
        out.writeVInt(totalShards);
        out.writeVInt(successfulShards);
        out.writeVInt(shardFailures.length);
        for (ShardSearchFailure shardSearchFailure : shardFailures) {
            shardSearchFailure.writeTo(out);
        }
        out.writeString(id);
        out.writeLong(creationTime);
```  

[1] https://github.com/opensearch-project/OpenSearch/pull/2745/files#diff-d9127de3d6633859920839481701bf460425591bdeb3e2a099579a38b1eacef1R103